### PR TITLE
feat(arm): AppServicePythonVersion - 82 check the 'python version' is the latest, if used to run the web app

### PR DESCRIPTION
**Pull request by RachelBorzi**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sast|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

feat(arm): add CKV_AZURE_82 to ensure that 'Python version' is the latest, if used to run the web app

### Description
I added a test that verifies whether the python version is latest if used to run the web app
### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/arm/checks/resource/AppServicePythonVersion.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            id=id,
+            categories=categories,
+            supported_resources=supported_resources,
+            missing_block_result=CheckResult.UNKNOWN)
+
+    def get_inspected_key(self) -> str:
+        return "properties/siteConfig/pythonVersion"
+
+    def get_expected_value(self) -> str:
+        return "3.4"
```
Comment (ChanochShayner) [inline checkov/arm/checks/resource/AppServicePythonVersion.py:23]: @tsmithv11 Should we use 3.12 here?

Comment (tsmithv11) [reply checkov/arm/checks/resource/AppServicePythonVersion.py:23]: We should accept 3.9-3.12 here (all not EOL or close to EOL). The TF policy is on a deprecated resource that can only go up to 3.4, but this one appears to be able to go up to 3.12.

Comment (ChanochShayner) [reply checkov/arm/checks/resource/AppServicePythonVersion.py]: So, @tsmithv11 correct me if I'm wrong, but it can be a Py check but a Yaml check, right?

Comment (tsmithv11) [reply checkov/arm/checks/resource/AppServicePythonVersion.py]: @ChanochShayner this is a single attribute inside a single resource, so a Python check will probably be better so we can have auto-fixes for it.

File: checkov/arm/checks/resource/AppServicePythonVersion.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            id=id,
+            categories=categories,
+            supported_resources=supported_resources,
+            missing_block_result=CheckResult.UNKNOWN)
+
+    def get_inspected_key(self) -> str:
+        return "properties/siteConfig/pythonVersion"
+
+    def get_expected_value(self) -> str:
+        return "3.10"
```
Comment (tsmithv11) [inline checkov/arm/checks/resource/AppServicePythonVersion.py:23]:
```suggestion
    def get_expected_values(self) -> List[Any]:
        return ["3.9", "3.10", "3.11", "3.12"]
```

**Review by ChanochShayner — APPROVED** [review]
Nice!

**Review by tsmithv11 — APPROVED** [review]
Great job 🚀

**Pull request merged**